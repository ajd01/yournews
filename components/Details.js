import React, { useContext } from 'react'
import { View, Text, Image, StyleSheet, Dimensions, Button, Linking, ScrollView } from 'react-native'
import Tools from './Tools'
import { Context } from '../store'

export default function Details () {
  const { store } = useContext(Context)

  const { selectedValue, data } = store

  const {
    author,
    publishedAt,
    title,
    urlToImage,
    description,
    content,
    url
  } = data[selectedValue]

  return (
    <ScrollView contentContainerStyle={{ paddingBottom: 200 }}>
      <View style={styles.detail}>
        <Text style={styles.title}>{title}</Text>

        <View style={{ flexDirection: 'row' }}>
          <View style={styles.half}>
            <Text style={styles.author}>{author}</Text>
            <Text style={styles.date}>{new Date(publishedAt).toDateString()}</Text>
          </View>
          <View style={styles.half}>
            <Tools id={selectedValue} />
          </View>
        </View>
        <Text style={styles.space}>
          {description}
        </Text>
        <Image style={styles.image} source={{ uri: urlToImage }} />
        <Text style={styles.space}>
          {content}
        </Text>
        <Button
          title='See more'
          onPress={() => {
            Linking.openURL(url)
          }}
        />
      </View>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  image: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').width,
    paddingBottom: 5,
    paddingTop: 20
  },
  title: {
    paddingTop: 10,
    fontSize: 25,
    fontWeight: 'bold',
    width: Dimensions.get('window').width
  },
  author: {
    fontSize: 15,
    width: Dimensions.get('window').width
  },
  date: {
    fontSize: 15,
    fontWeight: 'bold',
    width: Dimensions.get('window').width
  },
  half: {
    width: Dimensions.get('window').width / 2
  },
  space: {
    paddingTop: 20,
    paddingBottom: 20
  },
  button: {
    padding: 20,
    textAlign: 'center'
  },
  detail: {
    padding: 5
  }
})
