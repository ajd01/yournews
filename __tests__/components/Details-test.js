import React from 'react'
import Details from '../../components/Details'
import renderer from 'react-test-renderer'
import {
  Context,
  InitialState
} from '../../store'

jest.mock('../../components/Tools', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    }
  }
})

const value = {
  dispatch: jest.fn(),
  store: {
    ...InitialState,
    selectedValue: 0,
    data: [
      {
        author: 'Mockauthor',
        publishedAt: 'MockpublishedAt',
        title: 'Mocktitle',
        urlToImage: 'MockurlToImage',
        description: 'Mockdescription',
        content: 'Mockcontent',
        url: 'Mockurl'
      }
    ]
  }
}

describe('Details componente', () => {
  it('renders initial screen', () => {
    const tree = renderer.create(
      <Context.Provider value={value}>
        <Details />
      </Context.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
