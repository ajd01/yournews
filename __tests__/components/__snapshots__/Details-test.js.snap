// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details componente renders initial screen 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 200,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
            "paddingTop": 10,
            "width": 750,
          }
        }
      >
        Mocktitle
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "width": 375,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "width": 750,
              }
            }
          >
            Mockauthor
          </Text>
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "bold",
                "width": 750,
              }
            }
          >
            Invalid Date
          </Text>
        </View>
        <View
          style={
            Object {
              "width": 375,
            }
          }
        >
          <div />
        </View>
      </View>
      <Text
        style={
          Object {
            "paddingBottom": 20,
            "paddingTop": 20,
          }
        }
      >
        Mockdescription
      </Text>
      <Image
        source={
          Object {
            "uri": "MockurlToImage",
          }
        }
        style={
          Object {
            "height": 750,
            "paddingBottom": 5,
            "paddingTop": 20,
            "width": 750,
          }
        }
      />
      <Text
        style={
          Object {
            "paddingBottom": 20,
            "paddingTop": 20,
          }
        }
      >
        Mockcontent
      </Text>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            See more
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
